manifests/master/pdf_stack.pp

```diff
@@ -1,7 +1,7 @@
 # Dependencies for PDF rendering
 class classroom::master::pdf_stack {
 
-  $enabled = $classroom::offline ? {
+  $enabled = defined('$classroom::offline') and $classroom::offline ? {
```

Comment:
This feels like compounding the hacky solution. How about adding a parameter to the class instead?